src/main/java/com/bobocode/svydovets/context/AnnotationConfigurationApplicationContext.java

```diff
@@ -1,77 +1,136 @@
+            throw new NoSuchBeanDefinitionException(String.format("Bean with type %s is not found",
+              beanType.getSimpleName()));
         }
 
         return beanType.cast(beans.get(0));
     }
 
     @Override
-    public Object getBean(String name) throws NoSuchBeanDefinitionException {
-        return nameToBean.get(name);
+    public Object getBean(String beanName) throws NoSuchBeanDefinitionException {
+        Objects.requireNonNull(beanName, "The beanName cannot be null!");
+
+        if (StringUtils.isEmpty(beanName)) {
+            throw new IllegalArgumentException("Bean name is empty! Please specify the bean name.");
+        }
+
+        var bean = beanContainer.get(beanName);
+        if (Objects.isNull(bean)) {
+            throw new NoSuchBeanDefinitionException(String.format("Bean with name %s is not found", beanName));
+        }
+
+        return bean;
     }
 
     @Override
-    public <T> T getBean(String name, Class<T> beanType) throws NoSuchBeanDefinitionException {
-        var bean = nameToBean.get(name);
-        if (bean == null || !beanType.isInstance(beanType)) {
-            throw new NoSuchBeanDefinitionException();
+    public <T> T getBean(String beanName, Class<T> beanType) throws NoSuchBeanDefinitionException {
+        Objects.requireNonNull(beanName, "The beanName cannot be null!");
+        Objects.requireNonNull(beanType, "The beanType cannot be null!");
+        if (StringUtils.isEmpty(beanName)) {
+            throw new IllegalArgumentException("Bean name is empty! Please specify the bean name.");
         }
+
+        var bean = beanContainer.get(beanName);
+        if (bean == null) {
+            throw new NoSuchBeanDefinitionException(String.format("Bean with name %s is not found", beanName));
+        } else if (!ClassUtils.isAssignable(bean.getClass(), beanType)) {
+            throw new NoSuchBeanDefinitionException(
+              String.format("Bean with name %s is not an instance of the %s. %s is the instance of the %s",
+                beanName, beanType.getSimpleName(), beanName, bean.getClass().getSimpleName()));
+        }
+
         return beanType.cast(bean);
     }
 
     @Override
+    // TODO: should we return empty Map in case if we not find instances of beanType or throwException?
     public Map<String, Object> getBeans(Class<?> beanType) throws NoSuchBeanDefinitionException {
-        return  nameToBean.entrySet().stream()
-                .filter(entry -> beanType.isInstance(entry.getValue()))
-                .collect(Collectors.toMap(Map.Entry::getKey, entry -> beanType.cast(entry.getValue())));
+        return beanContainer.entrySet().stream()
+          .filter(entry -> beanType.isInstance(entry.getValue()))
+          .collect(Collectors.toMap(Map.Entry::getKey, entry -> beanType.cast(entry.getValue())));
     }
 
-    public void scan(String packageName) {
+    private void init(String packageName) {
+        scanAndCreate(packageName);
+        injectAll();
```

Comment:
Why do you put the injection logic here but not in the factory?

Reply:
because for current realization is better to do it here, when we start to refactor code according our agreement -> then we will move it to BeanFactory